DuoUniversal/Client.cs

```diff
@@ -128,9 +126,21 @@ public async Task<IdToken> ExchangeAuthorizationCodeFor2faResult(string duoCode,
                 throw new DuoException("Error exchanging the code for a 2fa token", e);
             }
 
+            return tokenResponse;
+        }
+
+        /// <summary>
+        /// Send the authorization code provided by Duo back to Duo in exchange for a full Duo response.
+        /// Will raise a DuoException if the username does not match the Id Token.
+        /// </summary>
+        /// <param name="duoCode">The one-time use code issued by Duo</param>
+        /// <returns>A TokenResponse authenticating the user and describing the authentication</returns>
+        private IdToken IdTokenFromResponse(TokenResponse tokenResponse, string username)

```

Comment:
Can we change the name of this to something like "ValidatedIdTokenFromResponse", I'll explain why below

Reply:
Maybe even just `ValidateResponse`?